easybuild/easyblocks/h/hpl.py

```diff
@@ -83,7 +83,13 @@ def build_step(self):
         Build with make and correct make options
         """
 
-        for envvar in ['MPICC', 'LIBLAPACK_MT', 'CPPFLAGS', 'LDFLAGS', 'CFLAGS']:
+        # Only use _MT version of LAPACK if openmp toolchainopt is True
+        if self.toolchain.options.get('openmp', False):
```

Comment:
openmp is, in my opinion, also used to specify usage of threaded libs or not.

Didn't know that that gimkl version explicitly used openmp. I copied a standard intel/foss hpl-2.2 eb file and built it with gimkl-2018b and it failed due to the use of LIBLAPACK_MT.

I'm of the opinion that the esayconfigs should not depend on how the easyblock is written, which it is without this change.

Reply:
And it will not affect a single HPL easyconfig in any bad way. You simply shouldn't use threaded blas/lapack without having -fopenmp (openmp:True)

Reply:
what about getting easybuilders/easybuild-framework/pull/2356 merged and start using that?

Reply:
@akesandgren Well, it will affect existing `HPL` easyconfigs, since the default for `openmp` is `False`, so you're effectively switching most `HPL` builds to use `LIBLAPACK` rather than `LIBLAPACK_MT`...

I agree with @migueldiascosta introducing `mt_blas_lapack` would be a better approach.

Reply:
Perhaps, but hpl.py would still need a change, which would have the same effect on old HPL easyconfigs. I.e., the use of LIBLAPACK_MT is simply wrong here.

Reply:
For what is worth, I agree with @boegel and @migueldiascosta. Using the new parameter makes more sense. But that seems blocked waiting for `$LIBFFT_MT` to be set in place. There are various references to it in the code though. @boegel what is the issue here?